pkg/controller/controller_utils.go

```diff
@@ -549,6 +550,13 @@ func GetPodFromTemplate(template *v1.PodTemplateSpec, parentObject runtime.Objec
 	}
 	prefix := getPodsPrefix(accessor.GetName())
```

Comment:
getPodsPrefix handles the length being exceeded by trimming off the trailing dash. what happens if `prefix` is already at its max and we then append an index?

Reply:
After experiments, it will be truncated later and append a random 5-character string to finally contain at most 63 characters, which means the appended index will also be discarded. I think better to truncate the prefix (63-5-len(maxIndex)-2(two dashes)) here to reserve the space for adding the suffix(`$index`)